Geometry/TrackerGeometryBuilder/src/TrackerGeomBuilderFromGeometricDet.cc

```diff
@@ -43,8 +43,14 @@ namespace {
 TrackerGeometry* TrackerGeomBuilderFromGeometricDet::build(const GeometricDet* gd,
                                                            const PTrackerParameters& ptp,
                                                            const TrackerTopology* tTopo) {
-  int BIG_PIX_PER_ROC_X = ptp.vpars[2];
-  int BIG_PIX_PER_ROC_Y = ptp.vpars[3];
+  if (ptp.vpars.size() != 6) {
+    edm::LogError("TrackerGeomBuilderFromGeometricDet")
+        << "Tracker parameters block from XMLs called vPars is expected to have 6 entries, but has " << ptp.vpars.size()
+        << " entrie(s).";
+  }
+
+  const int BIG_PIX_PER_ROC_X = ptp.vpars.at(2);
```

Comment:
if there is a `throw` above at line 47, there is no need for bound checking that comes at a price. I'd go for an `operator[]`